test/Microsoft.Azure.WebJobs.Host.UnitTests/Tables/TableFilterFormatterTests.cs

```diff
@@ -0,0 +1,191 @@
+            template = BindingTemplate.FromString("{Age} lt Age");
+            bindingData = new Dictionary<string, object>
+                {
+                    { "Age", "25 or true" }
+                };
+            ex = Assert.Throws<InvalidOperationException>(() => TableFilterFormatter.Format(template, (IReadOnlyDictionary<string, object>)bindingData));
+            Assert.Equal($"An invalid parameter value was specified for filter parameter 'Age'.", ex.Message);
+
+            // verify datetime day operator value
+            template = BindingTemplate.FromString("day({BirthDate}) eq 8");
+            bindingData = new Dictionary<string, object>
+                {
+                    { "BirthDate", "BirthDate) neq 8 or day(BirthDate" }
+                };
+            ex = Assert.Throws<InvalidOperationException>(() => TableFilterFormatter.Format(template, (IReadOnlyDictionary<string, object>)bindingData));
+            Assert.Equal($"An invalid parameter value was specified for filter parameter 'BirthDate'.", ex.Message);
+
+            // verify ambiguous values are handled correctly
+            template = BindingTemplate.FromString("Age eq '{Age}' or Age eq {Age}");
+            bindingData = new Dictionary<string, object>
+                {
+                    { "Age", "25 or true" }
+                };
+            ex = Assert.Throws<InvalidOperationException>(() => TableFilterFormatter.Format(template, (IReadOnlyDictionary<string, object>)bindingData));
+            Assert.Equal($"An invalid parameter value was specified for filter parameter 'Age'.", ex.Message);
+        }
+
+        [Fact]
+        public void Format_StringLiteral_ReturnsExpectedValue()
+        {
+            // verify single quotes are escaped
+            var bindingData = new Dictionary<string, object>
+                {
+                    { "Name", "x' or 'x' eq 'x" }
+                };
+            string result = GetFormattedValue("Name eq '{name}'", bindingData);
+            Assert.Equal("Name eq 'x'' or ''x'' eq ''x'", result);
+
+            // verify the same template reversed
+            bindingData = new Dictionary<string, object>
+                {
+                    { "Name", "x' or 'x' eq 'x" }
+                };
+            result = GetFormattedValue("'{name}' eq Name", bindingData);
+            Assert.Equal("'x'' or ''x'' eq ''x' eq Name", result);
+
+            // verify a multipart filter
+            bindingData = new Dictionary<string, object>
+                {
+                    { "Name", "O'Malley" },
+                    { "Age", "35"}
+                };
+            result = GetFormattedValue("(Age gt {Age}) and (Name eq '{Name}')", bindingData);
+            Assert.Equal("(Age gt 35) and (Name eq 'O''Malley')", result);
+
+            // datetime filter range
+            bindingData = new Dictionary<string, object>
+                {
+                    { "D1", "2000-01-10" },
```

Comment:
>"2000-01-10"  [](start = 28, length = 13)

One of these should be in a different date format just to exercise that we're actually doing the date formatting and just passing the strings through directly.